external-crates/move/crates/move-package-alt/src/schema/sha.rs

```diff
@@ -25,14 +28,25 @@ pub struct GitSha {
     inner: String,
 }
 
+impl GitSha {
+    /// Check if this is a full sha ({SHA_FULL_LENGTH} chars) or not
+    pub fn is_full_sha(&self) -> bool {
+        self.inner.len() == SHA_FULL_LENGTH
+    }
+
+    #[cfg(test)]
+    pub fn to_short_sha(&self) -> String {
+        self.inner[..7].to_string()
+    }
+}
+
 impl TryFrom<String> for GitSha {
     type Error = ShaError;
 
-    /// Check if the given string is a valid commit SHA, i.e., 40 character long with only
-    /// lowercase letters and digits
+    /// Check if the given string is a valid commit SHA, min 7 character, max 40 character long
+    /// with only lowercase letters and digits.
     fn try_from(input: String) -> ShaResult<Self> {
-        // implicit deps sha for SuiFlavor is 12 characters long
-        if input.len() != 40 && input.len() != 12 {
+        if input.len() > SHA_FULL_LENGTH || input.len() < SHA_MIN_LENGTH {
```

Comment:
We were discussing using the long shas everywhere except when we first pin the dependency; perhaps the `GitSha` type should only allow 40 character shas, which would force us to do the sha conversion correctly when we fetch the packages. I'm not sure how difficult that would be